src/fundus/scraping/pipeline.py

```diff
@@ -136,26 +138,28 @@ def build_unique_url_filter() -> URLFilter:
         # we use this custom variant of interleave_longest in order to be able
         # to delay the program flow between batches
         async def _async_article_interleave_longest() -> AsyncIterator[Article]:
-            batches: AsyncIterator[Tuple[Optional[Article], ...]] = async_zip_longest(*async_article_iterators)
+            # type: ignore[attr-defined]
+            batches: AsyncIterator[Tuple[Optional[Article], ...]] = aioitertools.itertools.zip_longest(
+                *async_article_iterators
+            )
             start_time = time.time()
             async for batch in batches:
                 basic_logger.debug(f"Batch took {time.time() - start_time} seconds")
-                for value in batch:
-                    if value is not None:
-                        response_cache.add(value.html.responded_url)
-                        yield value
+                for next_article in cast(Iterable[Article], filter(bool, batch)):
+                    response_cache.add(next_article.html.responded_url)
+                    yield next_article
                 if final_delay:
-                    await asyncio.sleep(final_delay() - time.time() + start_time)
+                    await asyncio.sleep(max(0.0, final_delay() - time.time() + start_time))
                 start_time = time.time()
 
-        i: int = 0
         if max_articles is None:
             max_articles = -1
-        async for article in _async_article_interleave_longest():
-            if i == max_articles:
-                break
+        async for article_index, article in aioitertools.builtins.enumerate(
+            _async_article_interleave_longest(), start=1
+        ):  # type:ignore[attr-defined]
```

Comment:
This type-ignore is properly leftover. (It also is an invalid type-ignore since there is a space missing after the `type:`)